dirty_cat/_check_dependencies.py

```diff
@@ -0,0 +1,36 @@
+"""
+Check versions from .requirements.txt to raise error at init.
+.requirements.txt is written from setup.cfg during setup.
+"""
+import re
+from importlib.metadata import version
+from pathlib import Path as _Path
+
+from ._config_requirements import deps
+
+
+def _check_pack_version(dep, package_name, required_version, sign):
+    installed_version = version(package_name)
+    if not eval(f"'{installed_version}' {sign} '{required_version}'"):
+        raise ImportError(
+            f"dirty_cat {_version} requires {dep}, but you have "
+            f"{package_name} {installed_version} which is incompatible."
+        )
+
+
+parent_dir = _Path(__file__).parent
+with open(parent_dir / "VERSION.txt") as _fh:
+    _version = _fh.read().strip()
+
+deps = deps["deps"].split("\n")
```

Comment:
I will replace this part with this scikit-learn utils `from sklearn.externals._packaging.version import parse` that deals with version parsing way better than my simple implementation.